src/main/java/org/openrewrite/java/migrate/util/IteratorNext.java

```diff
@@ -0,0 +1,71 @@
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.util;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesJavaVersion;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+public class IteratorNext extends Recipe {
+    private static final MethodMatcher ITERATOR_MATCHER = new MethodMatcher("java.util.Collection iterator()", true);
+    private static final MethodMatcher NEXT_MATCHER = new MethodMatcher("java.util.Iterator next()", true);
+
+    @Override
+    public String getDisplayName() {
+        return "Replace `iterator().next()` with `getFirst()`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace `SequencedCollection.iterator().next()` with `getFirst()`.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                Preconditions.and(
+                        new UsesJavaVersion<>(21),
+                        Preconditions.and(
+                                new UsesMethod<>(ITERATOR_MATCHER),
+                                new UsesMethod<>(NEXT_MATCHER)
+                        )
+                ),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+                        J.MethodInvocation mi = super.visitMethodInvocation(method, executionContext);
+                        if (NEXT_MATCHER.matches(mi) && ITERATOR_MATCHER.matches(mi.getSelect())) {
+                            Expression select = ((J.MethodInvocation) mi.getSelect()).getSelect();
+                            if (TypeUtils.isAssignableTo("java.util.SequencedCollection", select.getType())) {
+                                return mi.withSelect(select)
+                                        .withName(mi.getName().withSimpleName("getFirst"))
+                                        .withMethodType(mi.getMethodType().withName("getFirst"));
```

Comment:
I suspect that the prefix may get lost in this recipe.

Also, the type attribution will be slightly off here, because the method type will have `Iterator` as its declaring type.

Reply:
Hmm, I had lifted this snippet from the Simplify method chain recipe, but I'll double check, thanks!

Reply:
I've fixed the method type in 62be57c (I think); thanks for pointing that out! Here's a debugger view of the new situation 
![image](https://github.com/openrewrite/rewrite-migrate-java/assets/1027334/6a99dc73-7bf3-4cda-8c8e-3f8ed973d0d8)

Reply:
I'm wondering what the best approach would be for prefix handling; I don't really expect anyone to use comments between `.iterator()` and `.next()`. They might each be on a new line, but even then it seems more logical to collapse and not retain the whitespace or comments that were there before.

For reference: here's a test that asserts the _current_ behaviour.
```java
@Test
void retainPrefix() {
    rewriteRun(
      //language=java
      java(
        """
          import java.util.*;
              
          class Foo {
              void bar(List<String> collection) {
                  String first =
                    // Collection prefix
                    collection
                    // Iterator prefix
                    .iterator()
                    // Next prefix
                    .next();
              }
          }
          """,
        """
          import java.util.*;
                        
          class Foo {
              void bar(List<String> collection) {
                  String first =
                    // Collection prefix
                    collection
                    // Next prefix
                    .getFirst();
              }
          }
          """
      )
    );
}
```

Reply:
I am afraid the type attribution is still not quite correct. Now the method's return type is `Iterator`. I think it would be best if we could either search the owner type for the method (possibly involves searching the inheritance hierarchy upwards) or modify the type so that all its fields have the expected values (owner, name, type, parameters.

Regarding the prefix I will take a look tomorrow. Maybe the problem is just hypothetical.

Reply:
Since we are updating the `next()` invocation I believe this code is correct. As with many recipes we have situations where comments will be removed. So we could of course just not update the code if there are any comment that would end up getting removed. But that is maybe overkill.

Reply:
I liked the suggestion to not make changes when there's a risk of comments being lost; tried for a bit to get that to work, but [in my tests](https://github.com/openrewrite/rewrite-migrate-java/pull/339/commits/7e73ebb882d59cfec0edb48bd43dd8cca12c60cc) the comment on the iterator call is in the `JRightPadded.after` of the select before it, and that makes it inaccessible for now (at least from my understanding). Given that it's improbable I figured document it, such that we can show it was a conscious limitation with likely little chance of negative consequences.